packages/artifact/src/internal/download/download-artifact.ts

```diff
@@ -89,10 +92,40 @@ export async function streamExtractExternal(
         clearTimeout(timer)
         reject(error)
       })
-      .pipe(unzip.Extract({path: directory}))
-      .on('close', () => {
+      .pipe(unzip.Parse())
+      .on('entry', (entry: unzip.Entry) => {
+        const entryPath = path
+          .normalize(entry.path)
+          .replace(/^(\.\.(\/|\\|$))+/, '')
+        const fullPath = path.join(directory, entryPath)
+        core.debug(`Extracting artifact entry: ${fullPath}`)
+        if (entry.type === 'Directory') {
+          promises.push(resolveOrCreateDirectory(fullPath).then(() => {}))
+          entry.autodrain()
+        } else {
+          promises.push(
+            resolveOrCreateDirectory(path.dirname(fullPath)).then(() => {})
```

Comment:
Curious why would we create a directory using `fullPath` when it's not a directory?

Reply:
`path.dirname(fullPath)` only takes the directory to the file - not the file itself - and creates it if it doesn't exist

Reply:
Ah! Gotcha!